### DISCUSSION TITLE
Sound Blaster AudioPCI 128D emulation?

### DISCUSSION THREAD

**Discussion by bob12111** [joncampbell123/dosbox-x#4551] [Q&A]
Is there any way to get DOSBox X to reproduce Sound Blaster AudioPCI 128D for game sound/music?  I have tried installing the drivers but they don't seem to cooperate with DOSBox.  I would like to be able to have this as an option...

**Comment by rderooy** [comment]
In short no. Adding support for a new sound card takes a significant amount of effort.

**Reply by bob12111** [reply]
Well, I would like to see this added!

**Reply by rderooy** [reply]
Good on you. Download the source code and start programming!

**Reply by Torinde** [reply]
[86Box](https://github.com/86Box/86Box) and [PCem](https://github.com/sarah-walker-pcem/pcem) support that card, so you can "just" re-use code from there.

[Bochs](https://github.com/bochs-emu/Bochs) and [QEMU](https://gitlab.com/qemu-project/qemu) as well.

Related:
- #4851

**Reply by rderooy** [reply]
Yes, one can. And I'm sure there are several other sound cards that are much more interesting for something like DOSBox-X to emulate than the SB AudioPCI 128D. Something like the PAS16 perhaps? Or properly emulate the different GUS models? Or the AWE32/64?

**Reply by Torinde** [reply]
Agree. And those are also in various states present in other emulators.

**Comment by bob12111** [comment]
> [86Box](https://github.com/86Box/86Box) and [PCem](https://github.com/sarah-walker-pcem/pcem) support that card, so you can "just" re-use code from there.
Can they?  How would I get it to work on them?


> Yes, one can. And I'm sure there are several other sound cards that are much more interesting for something like DOSBox-X to emulate than the SB AudioPCI 128D. Something like the PAS16 perhaps? Or properly emulate the different GUS models? Or the AWE32/64?

It would be interesting to try those cards as well, just to see how the games sound with them...

**Comment by joncampbell123** [comment]
Many older DOS games support the PAS/PAS16 cards, that would be useful.

Since PAS cards also act like a Sound Blaster once configured, you could tie it into the Sound Blaster emulation. At least according to one PAS16 card I have, they act like a Sound Blaster 2.0 without the auto-init DMA.

**Reply by rderooy** [reply]
Indeed the PAS has two OPL2 chips and pre-dated the SB Pro 1.0, and there are a few games that support the PAS, but don't support SB Pro 1.0.
https://www.dosdays.co.uk/topics/Manufacturers/mediavision.php#PAS

The **PAS Plus** could actually be a good target, since it is backward compatible with the original PAS, AdLib and SB 1.5.

The PAS16 instead is based on a single OPL3 chip, like the SB Pro 2.0. It is not backward compatible with earlier PAS versions, only the AdLib and the SB1.5.
https://www.dosdays.co.uk/topics/Manufacturers/mediavision.php#PAS16

**Reply by Torinde** [reply]
Pro Audio Spectrum/16/Plus/Thunderboard are supported [by 625-894 games](https://www.mobygames.com/browse/games/dos/tic,1/full,50/), which is the highest number of all unsupported cards. 
[Vogons discussion](https://www.vogons.org/viewtopic.php?f=32&t=27943)

https://github.com/dosbox-staging/dosbox-staging/issues/3901
86Box emulates PAS16 - [partially](https://github.com/86Box/86Box/issues/836)?

**Reply by fire-lizard** [reply]
Not partially anymore, I have tested PAS16 emulation inside 86Box 4.2 with Windows 3.11 and several DOS games (Warlords II, Realms of Arkania II, Settlers II, SimCity 200, Pirates Gold, Lost Vikings) and all of them work perfectly with the card now.

**Reply by johnnovak** [reply]
> Pro Audio Spectrum/16/Plus/Thunderboard are supported [by 625-894 games](https://www.mobygames.com/browse/games/dos/tic,1/full,50/), which is the highest number of all unsupported cards. [Vogons discussion](https://www.vogons.org/viewtopic.php?f=32&t=27943)
> 
> [dosbox-staging/dosbox-staging#3901](https://github.com/dosbox-staging/dosbox-staging/issues/3901) 86Box emulates PAS16 - [partially](https://github.com/86Box/86Box/issues/836)?

That number means nothing. The Vogons discussion explains why there's almost nothing to be gained from emulating the PAS. There is no just need for it.
